File: Anul 2/Semestrul 2/Metode de proiectare si programare/labs/Tema 4-proiect Server-Client Csharp/proiectSCcsharp/angajatServices/IService.cs
using proiectSCcsharp.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace proiectSCcsharp.services
{
    public interface IService
    {
        Angajat login(Angajat employee, IObserver employeeObserver);
        void logout(Angajat employee);
        Angajat getAngajatByUsername(String username);
        List<Angajat> getAllAngajati();
        Spectacol getSpectacol(long id);
        Spectacol getSpectacolByAll(String artistName, String time, String place);
        List<Spectacol> getAllSpectacole();
        List<SpectacolDTO> getAllSpectacoleCautate(String date);
        void addBilet(String cumparatorName, long idSpectacol, int nrSeats);

    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 3-proiect Server-Client Java/proiectServerClientJava/Services/src/main/java/proiectServerClientJava/service/IService.java
package proiectServerClientJava.service;

import proiectServerClientJava.domain.Angajat;
import proiectServerClientJava.domain.Bilet;
import proiectServerClientJava.domain.Spectacol;
import proiectServerClientJava.domain.validators.ValidationException;

import java.time.LocalTime;

public interface IService {
    Angajat getAngajatByUsername(String username) throws MyException;
    Iterable<Angajat> getAllAngajati() throws MyException;
    Spectacol getSpectacol(Long id) throws MyException;
    Spectacol getSpectacolByAll(String artistName, LocalTime time, String place) throws MyException;
    Iterable<Spectacol> getAllSpectacole() throws MyException;
    void addBilet(String cumparatorName, long idSpectacol, int nrSeats) throws MyException, ValidationException;
    Angajat login(Angajat employee, IObserver client) throws Exception;
    void logout(Angajat employee, IObserver client) throws Exception;
}
